src/thunderbots/software/test/primitive/primitive.cpp

```diff
@@ -72,6 +74,57 @@ TEST(PrimitiveTest, convert_MovePrimitive_to_message_and_back_to_MovePrimitive)
     EXPECT_EQ(new_prim->getExtraBits(), std::vector<bool>());
 }
 
+TEST(PrimitiveTest, creat_MoveSpinPrimitive_from_message_test)
+{
+    const Point destination     = Point(-1, 4);
+    const Angle angular_vel     = AngularVelocity::ofRadians(0.54);
+    const unsigned int robot_id = 2U;
+
+    MoveSpinPrimitive movespin_prim =
+        MoveSpinPrimitive(robot_id, destination, angular_vel);
+
+    thunderbots_msgs::Primitive prim_message = movespin_prim.createMsg();
+
+    MoveSpinPrimitive new_prim = MoveSpinPrimitive(prim_message);
```

Comment:
```suggestion
    MoveSpinPrimitive new_prim = Primitive::createPrimitive(prim_message);
```

We want to use this generic constructor because it will use/test the new `if/else` statements added in `primitive.cpp`